### ISSUE TITLE
React integration ?

### ISSUE THREAD

**Issue by cristiandley** [pa7/heatmap.js#227]
Is there any future plan over React ?

**Comment by pa7** [comment]
Yes, a React Heatmap component is planned. If it's urgent and you want to give it a shot, the angular-heatmap plugin could be a good starting point:
https://github.com/pa7/heatmap.js/blob/master/plugins/angular-heatmap/angular-heatmap.js

**Comment by cristiandley** [comment]
Hey @pa7 im working on `react-heatmap` and `react-leaflet-heatmap`, the first one is completly working... i have a shit computer these days so im not able to test it completly but... if someone wants to take the job im happy. I will be posting `react-leaflet-heatmap` tomorrow...

`react-heatmap`

```javascript
import _ from 'lodash';
import h337 from 'heatmap.js';
import ReactDOM, { render } from 'react-dom';
import React, {Component, PropTypes} from 'react';

class ReactHeatmap extends Component {

  constructor(props, context) {
    super(props, context);
    this.state = { cfg: null };
  }

  componentDidMount(){
    const { style, data, config } = this.props;
    let _container = ReactDOM.findDOMNode(this);
    let defaultCfg = {
      width: style.width.replace('px','') || _container.offsetWidth,
      height: style.height.replace('px','') || _container.offsetHeight,
      container: _container
    };
    let _cfg = _.merge(defaultCfg, config);
    this.setState({ cfg: _cfg });
    const heatmapInstance = h337.create(cfg);
    heatmapInstance.setData(data);
  }


  render(){
    return (
      <div ref="react-heatmap"></div>
    );

  }

}

export default ReactHeatmap;
```

Have a nice time!

**Comment by cristiandley** [comment]
@pa7 im working on two solutions, one is `react-heatmap`, and the other one is `react-leaflet-heatmap`

[react-heatmap](https://gist.github.com/cristiandley/46157c292a97c99986f8bd98fcf89d0d)

[react-leaflet-heatmap](https://gist.github.com/cristiandley/46157c292a97c99986f8bd98fcf89d0d)

i got both working just like integrations but not like components... now im trying to export `react-heatmap` functions and working on `react-leaflet-heatmap`. There is something about the last one that i dont get... why is [this._origin](https://github.com/pa7/heatmap.js/blob/master/plugins/leaflet-heatmap/leaflet-heatmap.js#L210) called twice but newer used... is there a reason ?

Thanks !!!

**Comment by pa7** [comment]
@cristiandley nice catch! that's an artifact from a previous implementation and can be removed :)  are you planning to open a PR for the integrations?

**Comment by cristiandley** [comment]
@pa7 awesome !!! yes... i will be opening a PR as soon as i can :smile: (by the end of the week)

**Comment by cristiandley** [comment]
@pa7 are you there ?

**Comment by pa7** [comment]
@cristiandley sorry, didn't see the edit about es6. I'd prefer not adding a babel setup for the whole repo as I'm currently not using es6, however you could add it in the plugins subfolder e.g. in a folder  `react-heatmap`. Make sure to add the compiled code too so examples can reference it

**Comment by smoghal** [comment]
I ended up wrapping your library @pa7 in a react component.  It works quite nicely.  I'm more than happy to contribute it back to your repo or you can add a reference to mine.  Here's the [react implementation](https://github.com/smoghal/react-heatmap).

**Comment by Duoquote** [comment]
A while ago react-leaflet released v3 of react-leaflet, would like to see this plugin implementation in react-leaflet v3 too. The new api is completely different.
